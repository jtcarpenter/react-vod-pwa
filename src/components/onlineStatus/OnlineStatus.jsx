import React from 'react';
import styled from 'styled-components';
import * as copy from 'constants/copy';

export const StyledDiv = styled.div`
    position: fixed;
    bottom: 10%;
    width: 60%;
    margin: 0 20%;
    background-color: rgba(256, 256, 256, .5);
    border-radius: 15px;
    display: ${(props) => {
        return props.onlineStatus
            ? 'none'
            : 'block';
    }};
`;

export const StyledP = styled.p`
    text-align: center;
    color: ${(props) => props.theme.PRIMARY_DARK};
    font-weight: bold;
`

export function OnlineStatus({onlineStatus}) {
    return (
        <StyledDiv onlineStatus={onlineStatus}>
            <StyledP>
                {copy.OFFLINE_STATUS_MESSAGE}
            </StyledP>
        </StyledDiv>
    );
}

export default OnlineStatus;