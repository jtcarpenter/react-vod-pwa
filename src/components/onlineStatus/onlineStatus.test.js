import React from 'react';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';
import {OnlineStatus} from 'components/onlineStatus/OnlineStatus';

const mockOnlineStatus = (
    <OnlineStatus onlineStatus={true}/>
)

describe('OnlineStatus', () => {
    it('should render', () => {
        const wrapper = shallow(mockOnlineStatus);
        expect(wrapper.exists()).toBe(true);
    });

    it('should display correctly', () => {
        const onlineStatus = renderer.create(mockOnlineStatus);
        expect(onlineStatus.toJSON()).toMatchSnapshot();
    });
});