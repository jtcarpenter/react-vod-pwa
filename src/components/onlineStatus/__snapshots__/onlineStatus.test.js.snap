// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OnlineStatus should display correctly 1`] = `
<div
  className="sc-bdVaJa cJRoez"
>
  <p
    className="sc-bwzfXH aQnIu"
  >
    You are offline
  </p>
</div>
`;
